string mlnFiles
string db
string method
string engine
string output_filename
bool saveResults
string logic
string grammar
